functions/_middleware.js

```diff
@@ -0,0 +1,42 @@
+const cookieName = "a/a-test-cookie"
+const newHomepagePathName = "dtv=a"
+const throttle = 50 // 50% of traffic
+
+const abTest = async ({ request, next, env }) => {
+    const url = new URL(request.url)
+
+    // url is not on the homepage
+    if (url.pathname !== "/") {
+        return next();
+    }
+
+    const cookie = request.headers.get("cookie")
+
+    // client is in the `new` test
+    if (cookie && cookie.includes(`${cookieName}=new`)) {
+        url.search = newHomepagePathName
+        return env.ASSETS.fetch(url)
+    }
+
+    // client is in the `old` test
+    if (cookie && cookie.includes(`${cookieName}=current`)) {
```

Comment:
Hi @ali-hosseini-deriv 
Instead of 'new' and 'current' as cookie name, can we have 'control' as current and 'variant_1' for new. 

Also can we add support for multiple variants?

Reply:
@habib-deriv 
* For the naming i will do it right now.

* For the multiple variants i think we better focus on two variants at the same time.
the reason is that even now we cannot predict what percentage redirects to variant 1 or 2.
because we are redirecting randomly.
so it can be 40 to 60 percent or even 30 to 70 and so on!
so having more than 2 can cause less useful data.

Reply:
For 3 variants the traffic needs to be splitted 33% percent each for variant_a abd variant_b. And control gets 34%. Sure, we can add this as improvement if needed.